Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/ArrayDictionary.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+public class ArrayDictionary implements Dictionary {
+	private ArrayList <String> words=new ArrayList<String>();
+
+	//------------------------------------------
+	public ArrayDictionary(String filename){
+		try (BufferedReader r = new BufferedReader (new FileReader (filename));){
+			String temp = "";
+			do{
+				temp= r.readLine();
+				words.add(temp);
+			}while(temp!=null);
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();	//Shouldn't happen
+		} catch (IOException e) {
+			e.printStackTrace();	//In case of Empty file
+		}
+		assert(!words.isEmpty());
```

Comment:
Because I'd rather know something went wrong sooner rather than later. An emptdictionary will get you a null pointer the first time you try to load a string from it. I do agree that it's better to have a boolean static method from a loader class to tell, whether the dictionary is empty or not.